ros2/diagnostics/src/main.cpp

```diff
@@ -0,0 +1,140 @@
+#include <rclcpp/rclcpp.hpp>
+#include <std_msgs/msg/int16.hpp>
+#include <diagnostic_msgs/msg/diagnostic_array.hpp>
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+#include <diagnostic_msgs/msg/key_value.hpp>
+
+using namespace std::chrono_literals;
+using namespace std::placeholders;
+
+class DiagnosticsPublisher : public rclcpp::Node
+{
+public:
+    DiagnosticsPublisher() : 
+    Node("diagnostics_publisher"), 
+    timer_count_(0), 
+    last_msg_time_(now())
+    {
+        // Publisher for diagnostic array messages
+        diagnostics_pub_ = create_publisher<diagnostic_msgs::msg::DiagnosticArray>("/diagnostics", 10);
+
+        // Subscriber to the Int16 topic
+        topic_sub_ = create_subscription<std_msgs::msg::Int16>(
+            "/topic", 10, std::bind(&DiagnosticsPublisher::topic_callback, this, _1));
+
+        // Timer to update status based on internal counter
+        timer_ = create_wall_timer(100ms, std::bind(&DiagnosticsPublisher::timer_callback, this));
+
+        // Timer to check for staleness
+        staleness_timer_ = create_wall_timer(500ms, std::bind(&DiagnosticsPublisher::staleness_check_callback, this));
+    }
+
+private:
+    // Callback to get the value and time for the topic
+    void topic_callback(const std_msgs::msg::Int16::SharedPtr msg){
```

Comment:
Can you run clang-format or another formatter over this file so that it uses consistent code style?